ts-lib/tests/contract.test.ts

```diff
@@ -45,51 +38,31 @@ async function initContract(): Promise<EthKeysContract> {
   const near = await connect(config);
   const account = await near.account(accountId);
 
-  const contract = new Contract(account, contractName, {
-    viewMethods: ["get_key"],
-    changeMethods: ["set_key"],
-    useLocalViewExecution: false,
-  }) as unknown as EthKeysContract;
+  const contract = new KeyContract(contractName, account);
 
   return contract;
 }
 
-function encodeEthKey(key: string): string {
-  const bytes = Buffer.from(key.slice(2), "hex");
-  const encodedKey = bs58.encode(bytes);
-  return encodedKey;
-}
-
-function decodeEthKey(key: string): string {
-  const bytes = Buffer.from(bs58.decode(key));
-  return "0x" + bytes.toString("hex");
-}
-
 describe("EthKeys contract tests", () => {
-  let contract: EthKeysContract;
+  let contract: KeyContract;
 
   beforeAll(async () => {
     contract = await initContract();
   });
 
-  it("should generate ethWallet, encode and encrypt private key, set value on contract, retrieve, decrypt and match", async () => {
+  it("Base58 RoundTrip", async () => {
+    const keyManager = new Base58KeyManager(contract);
+
     // TODO - can we create not random?
-    let ethWallet = ethers.Wallet.createRandom();
-    let encodedEthKey = encodeEthKey(ethWallet.privateKey);
-    const { secret: encryptedKey, nonce } = create(
-      encodedEthKey,
-      keyPair.getPublicKey().toString(),
-      privateKey,
-    );
-    console.log("Encrypted Key", encryptedKey, nonce);
-    await contract.set_key({ encrypted_key: encryptedKey });
-    const retrievedKey = await contract.get_key({ account_id: accountId });
-    const decryptedKey = open(
-      retrievedKey!,
-      keyPair.getPublicKey().toString(),
-      privateKey,
-      nonce,
+    const ethWallet = ethers.Wallet.createRandom();
+
+    // TODO - include nonce on contract so we don't have to remember it.
+    const nonce = await keyManager.encryptAndSetKey(ethWallet, privateKey);
+
+    const decryptedKey = await keyManager.retrieveAndDecryptKey(
+      { accountId, privateKey },
+      nonce ? nonce : undefined,
```

Comment:
What is the type of `nonce`? This seems kinda redundant

Reply:
Unfortunately nonce is `string | null` so we are essentially converting `string | null` to `string | undefined` here. I would love to see a better way though.

The inconsistency here comes from using this `nonce?: string` notation which is `string | undefined` while the encryption library explicitly uses `string | null`.

Reply:
`nonce || undefined` seems more ergonomic, ideally at the return statement for `encryptAndSetKey`